xbmc/platform/freebsd/CPUInfoFreebsd.cpp

```diff
@@ -245,12 +245,16 @@ float CCPUInfoFreebsd::GetCPUFrequency()
 
 bool CCPUInfoFreebsd::GetTemperature(CTemperature& temperature)
 {
+  if (CCPUInfoPosix::GetTemperature(temperature))
```

Comment:
It looks a little confusing to me too about Posix.
Think the cleanest would be to have separated, e.g. `static CCPUInfo::TempertureByAdvancedUserSetCmd` (maybe you can come up with a better name)

So it can as code e.g:
```cpp
// as static on "xbmc/utils/CPUInfo.cpp"
CCPUInfo::TempertureByAdvancedUserSetCmd(CTemperature& temperature)
{
  temperature.SetValid(false);

  std::string cmd = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_cpuTempCmd;

  if (cmd.empty())
    return false;

  int value = {};
  char scale = {};

  auto p = popen(cmd.c_str(), "r");
  if (p)
  {
    int ret = fscanf(p, "%d %c", &value, &scale);
    pclose(p);

    if (ret < 2)
      return false;
  }

  if (scale == 'C' || scale == 'c')
    temperature = CTemperature::CreateFromCelsius(value);
  else if (scale == 'F' || scale == 'f')
    temperature = CTemperature::CreateFromFahrenheit(value);
  else
    return false;

  temperature.SetValid(true);

  return true;
}

// on "platform/posix/CPUInfoPosix.cpp"
bool CCPUInfoPosix::GetTemperature(CTemperature& temperature)
{
  // Within POSIX itself we not know how to get temperatur, so the user need to set inside advanced-settings.xml
  return CCPUInfo::TempertureByAdvancedUserSetCmd(temperature);
}

// xbmc/platform/linux/CPUInfoLinux.cpp
bool CCPUInfoLinux::GetTemperature(CTemperature& temperature)
{
  // Check user override the standard used way by advanced-settings.xml
  if (CCPUInfo::TempertureByAdvancedUserSetCmd(temperature))
    return true;

  if (m_tempPath.empty())
    return false;

  CSysfsPath path{m_tempPath};
  double value = path.Get<double>() / 1000.0;
  temperature = CTemperature::CreateFromCelsius(value);
  temperature.SetValid(true);
  return true;
}
```

Reply:
Updated, thanks.